src/main/java/com/o19s/es/ltr/feature/PrebuiltFeature.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright [2017] Wikimedia Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.o19s.es.ltr.feature;
+
+import org.apache.lucene.search.Query;
+import org.elasticsearch.index.query.QueryShardContext;
+
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * A prebuilt featured query, needed by query builders
+ * that provides the query itself.
+ */
+public class PrebuiltFeature implements Feature {
+    private final String name;
+    private final Query query;
+
+    public PrebuiltFeature(String name, Query query) {
+        this.name = name;
```

Comment:
Yes for now we still have to support null names, adding `@Nullable` from elastic